tasks/main.yml

```diff
@@ -23,6 +23,82 @@
   ufw: rule=allow port={{ item }} proto=tcp
   with_items:
     - 3306
+#
+# Parameters to set the maximum number of open files
+#
+- name: Creating directories for Mysql files limit configurations
+  file:
+    path: "{{ item }}"
+    state: directory
+    mode: 0755
+    owner: root
+    group: root
+  with_items:
+    - /etc/mysql/conf.d/
+    - /etc/systemd/system/mysql.service.d/
+
+# 
+# The limits of processes started by systemd/upstart are governed by systemd/upstart
+# configuration files and not from security/limits.d
+# Ubuntu 14.04 uses upstart, Ubuntu 16.04 systemd.
+- name: Set open files limit in mysql through systemd
+  blockinfile:
+    dest: /etc/systemd/system/mysql.service.d/override.conf
+    create: yes
+    owner: root
+    group: root
+    block: |
+      # cf https://duntuk.com/how-raise-ulimit-open-files-and-mysql-openfileslimit
+      [Service]
+      LimitNOFILE={{ MYSQL_OPEN_FILES_LIMIT }}
+      LimitNPROC={{ MYSQL_PROCESSES_LIMIT }}
+#
+# The "upstart" version is commented out because the presence of this small
+#  /etc/init/mysql.conf prevents the real one from being installed in a later step
+# (it ends in /etc/init/mysql.conf.dpkg-dist instead)
+#
+# - name: Set open files limit in mysql through upstart
+#   blockinfile:
+#     dest: /etc/init/mysql.conf
+#     create: yes
+#     owner: root
+#     group: root
+#     block: |
+#       # cf https://duntuk.com/how-raise-ulimit-open-files-and-mysql-openfileslimit
+#       limit nofile {{ MYSQL_OPEN_FILES_LIMIT }} {{ MYSQL_OPEN_FILES_LIMIT }}
+#       limit nproc {{ MYSQL_PROCESSES_LIMIT }} {{ MYSQL_PROCESSES_LIMIT }}
+
+# Settings in security/limits.d affect only processes started from login (i.e.
+# from an interactive shell). They aren't needed to change the limits of the mysql
+# process.
+# After being set, these settings can be confirmed by doing:
+# cat /proc/SOME_PID/limits
+# - name: Set open files limit in security/limits.d
+#   blockinfile:
+#     dest: /etc/security/limits.d/90-nproc.conf
+#     create: yes
+#     owner: root
+#     group: root
+#     block: |
+#       # cf https://duntuk.com/how-raise-ulimit-open-files-and-mysql-openfileslimit
+#       * soft nofile {{ MYSQL_OPEN_FILES_LIMIT }}
+#       * hard nofile {{ MYSQL_OPEN_FILES_LIMIT }}
+#       * soft nproc {{ MYSQL_PROCESSES_LIMIT }}
+#       * hard nproc {{ MYSQL_PROCESSES_LIMIT }}
+#       root soft nproc unlimited
+
+# After being set, this can be verified by logging in to mysql and doing:
+#    show global variables like 'open%';
+- name: Set open files limit in MySQL config
+  blockinfile:
+    dest: /etc/mysql/conf.d/open_files_limit.cnf
```

Comment:
This file does not seem to be picked up by the daemon, but neither does it seem necessary.  When starting, the daemon automatically uses the values from the ulimits, so that's all we need to change.  The only reason we needed to do this for the production daemon is that it was already running, so this whole task can be deleted.